package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh

```diff
@@ -667,10 +667,13 @@ mac80211_prepare_vif() {
 
 	json_select ..
 
-	[ -n "$macaddr" ] || {
+	if [ -z "$macaddr" ]; then
 		macaddr="$(mac80211_generate_mac $phy)"
 		macidx="$(($macidx + 1))"
-	}
+	elif [ "$macaddr" = 'random' ]; then
+		[ $(command -v 'macaddr_random') ] || source '/lib/functions/system.sh'
```

Comment:
no need to check whether it is imported already, just do it

dot instead of source
`. /lib/functions/system.sh`